akka-stream/src/main/scala/akka/stream/Attributes.scala

```diff
@@ -365,10 +364,30 @@ object Attributes {
 object ActorAttributes {
   import Attributes._
   final case class Dispatcher(dispatcher: String) extends MandatoryAttribute
+
+  object Dispatcher {
+    /**
+     * Resolves the dispatcher's name with a fallback to the default blocking IO dispatcher.
+     * Note that `IODispatcher.dispatcher` is not used here as the config used to create [[ActorMaterializerSettings]]
+     * is not easily accessible, instead the name is taken from `settings.blockingIoDispatcher`
+     */
+    def resolve(attributes: Attributes, settings: ActorMaterializerSettings): String =
+      attributes.mandatoryAttribute[Dispatcher] match {
+        case IODispatcher           ⇒ settings.blockingIoDispatcher
+        case Dispatcher(dispatcher) ⇒ dispatcher
+      }
+
+    /**
+     * Resolves the dispatcher name with a fallback to the default blocking IO dispatcher.
+     */
+    def resolve(context: MaterializationContext): String =
```

Comment:
`MaterializationContext` is `@InternalApi` so if this is needed the method `resolve` must also be `@InternalApi`